app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -934,6 +935,19 @@ public boolean isFullScreen() {
         return mViewModel.getIsFullscreen().getValue().get();
     }
 
+    public void setKioskMode(boolean isKioskMode) {
+        if (mViewModel.getIsKioskMode().getValue().get() != isKioskMode) {
```

Comment:
I personally prefer to early return here if getKioskMode()==isKioskMode. Then the rest of the code won't need this extra indentation level.

Reply:
Done.